diff --git a/src/BLE.cpp b/src/BLE.cpp
index 18a40cd..a916557 100644
--- a/src/BLE.cpp
+++ b/src/BLE.cpp
@@ -38,7 +38,7 @@ public:
 };
 
 void otaCallback::onWrite(BLECharacteristic *pCharacteristic) {
-  std::string rxData = pCharacteristic->getValue();
+  String rxData = pCharacteristic->getValue();
   if (!updateFlag) { // If it's the first packet of OTA since bootup, begin OTA
     // Serial.println("Begin FW Update");
     esp_ota_begin(esp_ota_get_next_update_partition(NULL), OTA_SIZE_UNKNOWN,
diff --git a/src/config.h b/src/config.h
index 108dc39..c67a1cb 100644
--- a/src/config.h
+++ b/src/config.h
@@ -1,6 +1,8 @@
 #ifndef CONFIG_H
 #define CONFIG_H
 
+#include "driver/gpio.h"
+
 //pins
 #if !defined(ARDUINO_WATCHY_V10) && !defined(ARDUINO_WATCHY_V15) && !defined(ARDUINO_WATCHY_V20)
 
@@ -9,7 +11,7 @@
 
 //Change to your board version
 #define ARDUINO_WATCHY_V20
-
+#endif
 #define MENU_BTN_PIN 26
 #define BACK_BTN_PIN 25
 #define DOWN_BTN_PIN 4
@@ -45,7 +47,7 @@
 #define ACC_INT_MASK  GPIO_SEL_14
 #define BTN_PIN_MASK  MENU_BTN_MASK|BACK_BTN_MASK|UP_BTN_MASK|DOWN_BTN_MASK
 
-#endif
+
 
 //display
 #define DISPLAY_WIDTH 200
